### PR TITLE
lua: Allow to set header entry as table to httpCall and respond APIs

### LINKED ISSUE #7742: Lua EnvoyFilter set multiple set-cookie header in request_handle functions
*Title*: *Set mutiple `Set-Cookie` header within a Lua EnvoyFilter as reponse*
_istio verision:_ v1.2.2

I tried to concat with `, ` like
```
headers["set-cookie"] = "test=123; Path=/, foo=bar; Path=/" 
```
This will be send but not unfolded.

Also tied using a table (like Nginx does it) no luck -> `got table, sting expected`.

### PR DESCRIPTION
**Pull request by dio**
Description: Extends `httpCall()` and `respond()` APIs to accept headers with values that can be strings or table of strings. 

Risk Level: Low
Testing: Modified the existing unit tests
Docs Changes: Added
Release Notes: Added
Fixes #7742

Signed-off-by: Dhi Aurrahman <dio@tetrate.io>

### CODE REVIEW

**Comment by dio** [comment]
@mattklein123 do we want to return header values of the same header entry key as a table when we do `headers:get()` as well, should it be filtered for non-":" and predefined headers?

**Comment by mattklein123** [comment]
> @mattklein123 do we want to return header values of the same header entry key as a table when we do headers:get() as well, should it be filtered for non-":" and predefined headers?

I'm concerned this might break existing scripts? WDYT? Perhaps we don't do it for now or provide some type of API ability to return it this way?

**Comment by mattklein123** [comment]
@dio please let me know when this is no longer WIP. Thank you!

/wait

**Comment by dio** [comment]
@mattklein123 yes, think this is ready for review. Thank you!

While for getting multiple values from a header name can be done via `:pairs` for now.

File: docs/root/intro/version_history.rst [hunk 1]
```diff
@@ -16,6 +16,7 @@ Version history
 * http: added the ability to reject HTTP/1.1 requests with invalid HTTP header values, using the runtime feature `envoy.reloadable_features.strict_header_validation`.
 * http: added the ability to :ref:`merge adjacent slashes<envoy_api_field_config.filter.network.http_connection_manager.v2.HttpConnectionManager.merge_slashes>` in the path.
 * listeners: added :ref:`HTTP inspector listener filter <config_listener_filters_http_inspector>`.
+* lua: extended `httpCall()` and `respond()` APIs to accept headers with entry value that can be string or table of strings.
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]:
```suggestion
* lua: extended `httpCall()` and `respond()` APIs to accept headers with entry values that can be a string or table of strings.
```

**Comment by dio** [comment]
Oops. Accepting suggestion hurts DCO. Did force pushing. Sorry.

File: docs/root/intro/version_history.rst (hunk 1 above)
Comment (dio) [reply docs/root/intro/version_history.rst]: Done. Thank you!

**Pull request merged**